scripts/build.sh

```diff
@@ -4,8 +4,70 @@ set -e
 
 CLEANSLATE="node_modules/cleanslate/docs/files/cleanslate.css"
 
+COLOR_RESET=$(tput sgr0 2>/dev/null)
+COLOR_BOLD=$(tput bold 2>/dev/null)
+COLOR_BAD=$(tput setaf 1 2>/dev/null)
+COLOR_GOOD=$(tput setaf 2 2>/dev/null)
+
+WIN_PYTHON_CMD="py -3"
+WIN_PREREQUISITES="tput printf cygpath which npm"
+
+NATIVE_BIN_DIR="native"
+WIN_COMPILE_NATIVE_BIN_SOURCE="${NATIVE_BIN_DIR}/native_main.py"
+WIN_COMPILE_NATIVE_BIN_OUTPUT="${NATIVE_BIN_DIR}/native_main.exe"
+WIN_COMPILE_NATIVE_BIN_SIG_FILE="${NATIVE_BIN_DIR}/native_main.exe.sig"
+WIN_COMPILE_NATIVE_BIN_HASH_FILE="${NATIVE_BIN_DIR}/native_main.exe.sha256"
+WIN_NATIVE_BIN_INSTALLER="${NATIVE_BIN_DIR}/win_install.ps1"
+
+##
+## This should be re-adjusted by the 'cmcaine/tridactyl' repo
+## maintainers when GPG signing is enabled
+##
+
+WIN_COMPILE_NATIVE_BIN_GPG2_SIGNER="gsbabil@gmail.com"
+
+if [ ! -z "${PYINSTALLER}" ] \
+  && [ "${PYINSTALLER}" = "1" ]; then
+  WIN_COMPILE_NATIVE_BIN="True"
+  WIN_PREREQUISITES="${WIN_PREREQUISITES} pyinstaller"
+else
+  WIN_COMPILE_NATIVE_BIN="False"
+fi
+
+stripWhitespace() {
+  local input="$@"
+  echo "${input}" | tr -d "[:space:]"
+}
+
+colorEcho() {
+  local str="$1"
+  local color="${COLOR_GOOD}${COLOR_BOLD}"
+
+  if [ ! -z "$2" ] \
+    && [ "$(stripWhitespace "$2")" = "alert" ]; then
+    color="${COLOR_BAD}${COLOR_BOLD}"
+  fi
+
+  echo -e "${color}${str}${COLOR_RESET}"
+}
+
+checkWindowsPrerequisite() {
+  local bin_name="$1"
+  local bin_loc=$("${WIN_WHICH_BIN_PATH}" \
+    "${bin_name}" 2>/dev/null)
+
+  if [ -z "${bin_loc}" ] \
+    || [ ! -f "${bin_loc}" ]; then
+    echo "    - '$1' not found, quitting ..."
+    exit -1
+  else
+    echo "    - '${bin_name}' found at ${bin_loc}"
+  fi
+}
+
```

Comment:
Would you mind moving most of this out to another script? The build script is one of the most important bits of Tridactyl so I want to keep the amount of noise to a minimum to help debugging. You could just do `if windows; run build.windows.sh and exit; fi` etc.